org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/component/ProcessInfoDialog.java

```diff
@@ -216,6 +242,63 @@ public void onEvent(Event event) throws Exception {
 		if (event.getTarget() == btnOk) {
 			this.detach();
 		}
+		if(event.getTarget() == btnPrint) {
+			Clients.showBusy(Msg.getMsg(Env.getCtx(), "Processing"));
+			Clients.response(new AuEcho(ProcessInfoDialog.this, "onPrint", null));
+		}
+	}
+	
+	/**
+	 * On Print
+	 */
+	public void onPrint() {
+		// Loop through all items
+		List<File> pdfList = new ArrayList<File>();
+		for (int i = 0; i < m_logs.length; i++)
+		{
+			int recordID = m_logs[i].getRecord_ID();
+			ReportEngine re = null;
+			
+			if(recordID > 0) {
+				re = ReportEngine.get (Env.getCtx(), ReportEngine.getReportEngineType(m_logs[i].getAD_Table_ID()), recordID);

```

Comment:
the safe thing to do here is `int reportEngineType = ReportEngine.getReportEngineType(m_logs[i].getAD_Table_ID()); if (reportEngineType >= 0) { ...}`